src/api/channels/channels.rs

```diff
@@ -43,12 +43,12 @@ impl Channel {
     /// # Returns
     ///
     /// A `Result` that contains a `ChorusLibError` if an error occurred during the request, or `()` if the request was successful.
-    pub async fn delete(self, user: &mut UserMeta) -> ChorusResult<()> {
+    pub async fn delete(channel_id: Snowflake, user: &mut UserMeta) -> ChorusResult<()> {
```

Comment:
Please revert the change to the functions signature, the rest looks good! Thanks once again for this PR :)